iceoryx_examples/iceperf/uds.cpp

```diff
@@ -46,6 +47,8 @@ void UDS::init() noexcept
     // initialize the sockAddr data structure with the provided name
     memset(&m_sockAddrPublisher, 0, sizeof(m_sockAddrPublisher));
     m_sockAddrPublisher.sun_family = AF_LOCAL;
+    const uint64_t maxDestinationLength = iox::cxx::strlen2(m_sockAddrPublisher.sun_path);
+    assert(maxDestinationLength >= m_publisherName.size() && "Socketname too large!");
```

Comment:
Is an assert here really sufficient? I think this check needs to be there in release mode as well or do we have full control over what is passed via m_publisherName? 

If yes, an assert suffices.

Reply:
It's just a tests, therefore I took the easiest route, but I can also add a "std::terminate there".

@budrus why wasn't the `UnixDomainSocket` class used for the example? Because it doesn't support zero copy at the moment? I'd like to add a zero copy interface to if when I port it away from std::string

Reply:
@elBoberido I wanted to use the UDS as raw as possible. Otherwise people could argue that we loose some performance because of our abstraction layers

Reply:
@MatthiasKillat it's the example with the benchmark, not a test like I mentioned before, and we have full control, so this is just to appease my inner chicken

Reply:
@elBoberido I would prefer to use `cxx::Expects` here since `assert` is the old C way and `cxx::Expects` provides you with a more detailed output like 
  - file 
  - line
  - function
  - condition 

when something goes wrong.

Additionally, it is more expressive..

Reply:
done